src/Compilers/VisualBasic/Portable/Emit/PEModuleBuilder.vb

```diff
@@ -443,23 +444,32 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Emit
                                     ' all exported types precede forwarded types, therefore contender cannot be a forwarded type.
                                     Debug.Assert(contender.ContainingAssembly Is sourceAssembly)
 
-                                    context.Diagnostics.Add(New VBDiagnostic(ErrorFactory.ErrorInfo(ERRID.ERR_ExportedTypesConflict,
-                                                                                                exportedType, exportedType.ContainingModule,
-                                                                                                contender, contender.ContainingModule),
-                                        NoLocation.Singleton))
+                                    context.Diagnostics.Add(New VBDiagnostic(ErrorFactory.ErrorInfo(
+                                                                                ERRID.ERR_ExportedTypesConflict,
+                                                                                CustomSymbolDisplayFormatter.DefaultErrorFormat(exportedType),
+                                                                                CustomSymbolDisplayFormatter.DefaultErrorFormat(exportedType.ContainingModule),
+                                                                                CustomSymbolDisplayFormatter.DefaultErrorFormat(contender),
+                                                                                CustomSymbolDisplayFormatter.DefaultErrorFormat(contender.ContainingModule)),
+                                                                             NoLocation.Singleton))
                                 Else
                                     If contender.ContainingAssembly Is sourceAssembly Then
                                         ' Forwarded type conflicts with exported type
-                                        context.Diagnostics.Add(New VBDiagnostic(ErrorFactory.ErrorInfo(ERRID.ERR_ForwardedTypeConflictsWithExportedType,
-                                                                                                    exportedType, exportedType.ContainingAssembly,
-                                                                                                    contender, contender.ContainingModule),
-                                            NoLocation.Singleton))
+                                        context.Diagnostics.Add(New VBDiagnostic(ErrorFactory.ErrorInfo(
+                                                                                    ERRID.ERR_ForwardedTypeConflictsWithExportedType,
+                                                                                    CustomSymbolDisplayFormatter.DefaultErrorFormat(exportedType),
+                                                                                    CustomSymbolDisplayFormatter.DefaultErrorFormat(exportedType.ContainingAssembly),
```

Comment:
It doesn't look like we need to do this for AssemblySymbols, ConvertSymbolToString always formats them the same way regardless of the error number. The goal, as I understand it, is to get rid of special cases, which are based on error numbers.